app/gui2/src/components/GraphEditor/GraphVisualization.vue

```diff
@@ -273,10 +295,43 @@ const effectiveVisualization = computed(() => {
   }
   return visualization.value
 })
+
+const root = ref<HTMLElement>()
+
+const keydownHandler = visualizationBindings.handler({
+  nextType: () => {
+    if (props.isFocused || focusIsIn(root.value)) {
+      const currentIndex = allTypes.value.findIndex((type) => visIdentifierEquals(type, currentType.value))
+      const nextIndex = (currentIndex + 1) % allTypes.value.length
+      emit('update:id', allTypes.value[nextIndex]!)
+    } else {
+      return false
+    }
+  },
+  toggleFullscreen: () => {
+    if (props.isFocused || focusIsIn(root.value)) {
+      emit('update:fullscreen', !props.isFullscreen)
+    } else {
+      return false
+    }
+  },
+  exitFullscreen: () => {
+    if (props.isFullscreen) {
+      emit('update:fullscreen', false)
+    } else {
+      return false
+    }
+  }
+})
+
+
+useEvent(window, 'keydown', (event) => keydownHandler(event))
```

Comment:
If GraphEditor handles `nextType` event, then there's no need for every visualization listen at window level. Instead we could attach handler to it's div (but then let's make sure visualization is focused after going fullscreen).